src/compiler/crystal/codegen/primitives.cr

```diff
@@ -274,6 +274,7 @@ class Crystal::CodeGenVisitor
     op_overflow = new_block "overflow"
     op_normal = new_block "normal"
 
+    overflow_condition = builder.call(llvm_expect_i1_fun, [overflow_condition, llvm_false])
```

Comment:
Cool! I wonder what difference it makes in terms of performance, but it's nice that LLVM has this feature.